gem/gem.py

```diff
@@ -1041,3 +1041,12 @@ def as_gem(expr):
         return Literal(expr)
     else:
         raise ValueError("Do not know how to convert %r to GEM" % expr)
+
+
+def collect_node_type(expressions, node_type):
+    """Collects Node objects of type node_type in expressions."""
+    vset = set()
+    for node in traversal(expressions):
+        if isinstance(node, node_type):
+            vset.add(node)
+    return tuple(vset)
```

Comment:
Call this `extract_type` and `traversal` is already unique so:
```
tuple(node for node in traversal(expressions) if isinstance(node, klass))
```

is sufficient.

Reply:
Thanks! I'll fix.